### PR TITLE
refactor: remove `ServiceInfo` from HTTP Connection Manager filter gen

### PR DESCRIPTION
**Pull request by nareddyt**
- Move HTTP connection manager into its own filter generator
- Remove the dependence on `ServiceInfo` from HTTP connection manager
- Remove `LocalReplyConfig`. It is not needed in ESPv2, we will override this filter gen in google3.
- Refactor tracing - main change is to remove creation of metadata fetcher from tracing library. Tracing project ID is dependency injected by config manager.
- Refactor tracing - move tracing options into their own struct

### CODE REVIEW

File: src/go/configgenerator/filtergen/http_connection_manager.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+const (
+	// HTTPConnectionManagerFilterName is the Envoy filter name for debug logging.
+	HTTPConnectionManagerFilterName = "envoy.filters.network.http_connection_manager"
+)
+
+type HTTPConnectionManagerGenerator struct {
```
Comment (TAOXUY) [inline src/go/configgenerator/filtergen/http_connection_manager.go:42]: Why not put HCM under `configgenerator` folder not `configgenerator/filter` folder? Will we have different HCM HCM configgenrator inside g3? 

HCM containing all http filters is a one-level lower than those in `filter` folder so it is wired to put them together.

Comment (nareddyt) [reply src/go/configgenerator/filtergen/http_connection_manager.go:42]: We will have a different HCM in google3. Remember you added custom local reply config for our use case? I removed that from ESPv2, but we will override it internally.

I would like to reuse some of the test utilities, like `filtergentest`. Not sure if moving it out of `filtergen` folder is good idea then.

In the future if we have more network filters, I can separate `filtergen` into `httpfiltergen` and `networkfiltergen`

**Comment by nareddyt** [comment]
PTAL

**Review by shuoyang2016 — APPROVED** [review]

**Review by TAOXUY — APPROVED** [review]

**Pull request merged**